client/state/push-notifications/selectors.js

```diff
@@ -0,0 +1,73 @@
+
+export const isApiReady = ( state ) => !! state.pushNotifications.settings.apiReady;
+export const isAuthorized = ( state ) => !! state.pushNotifications.settings.authorized;
+export const isBlocked = ( state ) => !! state.pushNotifications.settings.blocked;
+export const isEnabled = ( state ) => !! state.pushNotifications.settings.enabled;
+export const isNoticeDismissed = ( state ) => !! state.pushNotifications.settings.dismissedNotice;
+export const isShowingUnblockInstructions = ( state ) => !! state.pushNotifications.settings.showingUnblockInstructions;
+export const getSavedSubscription = ( state ) => state.pushNotifications.settings.subscription;
+export const getSavedWPCOMSubscription = ( state ) => state.pushNotifications.settings.wpcomSubscription;
+export const getLastUpdated = ( state ) => state.pushNotifications.settings.lastUpdated;
+
+export function isPushNotificationsSupported() {
+	return (
+		isServiceWorkerSupported() &&
+		'showNotification' in window.ServiceWorkerRegistration.prototype &&
+		'PushManager' in window
+	);
+}
+
+export function isServiceWorkerSupported() {
+	return (
+		'serviceWorker' in window.navigator &&
+		'ServiceWorkerRegistration' in window
+	);
+}
+
+export function isPushNotificationsDenied() {
+	return (
+		( ! ( 'Notification' in window ) ) ||
+		'denied' === window.Notification.permission
+	);
+}
+
+export function getDeviceId( state ) {
+	const subscription = getSavedWPCOMSubscription( state );
+	if ( ! subscription ) {
+		return;
+	}
+	return subscription.ID;
+}
+
+export function isNoticeVisible( state ) {
+	return (
+		isApiReady( state ) &&
+		! isAuthorized( state ) &&
+		! isEnabled( state ) &&
+		! isNoticeDismissed( state )
+	);
+}
+
+export function isSubscribed( state ) {
```

Comment:
This isn't really used outside this file...can probably delete it and just check the subscription in `getStatus`.